codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -45,10 +47,327 @@ pub enum Ecosystem {
+                    _ => None,
+                })
+                .unwrap();
+            assert_eq!(
+                interpolated_config_string,
+                r#"
+networks:
+  - id: 0
+    start_block: 0
+"#
+            );
+        }
+
+        #[test]
+        fn test_interpolate_config_variables_with_missing_env() {
+            let config_string = r#"
+networks:
+  - id: ${ENVIO_NETWORK_ID}
+    rpc_config:
+      url: https://eth.com?api_key=${ENVIO_ETH_API_KEY}
+"#;
+            let interpolated_config_string =
+                super::interpolate_config_variables(config_string.to_string(), |name| match name {
+                    "ENVIO_NETWORK_ID" => Some("0".to_string()),
+                    _ => None,
+                })
+                .unwrap_err();
+            assert_eq!(
+                interpolated_config_string.to_string(),
+                r#"Failed to interpolate variables into your config file. Environment variables are not preset: ENVIO_ETH_API_KEY"#
+            );
+        }
+
+        #[test]
+        fn test_interpolate_config_variables_with_invalid_captures_and_missing_env() {
+            let config_string = r#"
+networks:
+  - id: ${ENVIO_NETWORK_ID}
+    rpc_config:
+      url: ${My RPC URL}?api_key=${}
+"#;
+            let interpolated_config_string =
+                super::interpolate_config_variables(config_string.to_string(), |name| match name {
+                    "ENVIO_NETWORK_ID" => Some("0".to_string()),
+                    _ => None,
+                })
+                .unwrap_err();
+            assert_eq!(
+                interpolated_config_string.to_string(),
+                r#"Failed to interpolate variables into your config file. Invalid environment variables are preset: "My RPC URL", """#
+            );
+        }
+
+        #[test]
+        fn test_interpolate_config_variables_with_different_substituations() {
+            let config_string = r#"
+DirectSubstitution with existing env: "${EXISTING_ENV}"
+DefaultForMissing with existing env: "${EXISTING_ENV-default}"
+DefaultForMissing with existing env and many dashes: "${EXISTING_ENV----:---}"
+DefaultForMissing with missing env: "${MISSING_ENV-default}"
+DefaultForMissing with missing env and many dashes: "${MISSING_ENV----:---}"
+DefaultForMissing with missing env and empty default: "${MISSING_ENV-}"
+DefaultForMissingAndEmpty with existing env: "${EXISTING_ENV:-default}"
+DefaultForMissingAndEmpty with existing env and many dashes: "${EXISTING_ENV:----:---}"
+DefaultForMissingAndEmpty with missing env: "${MISSING_ENV:-default}"
+DefaultForMissingAndEmpty with missing env and many dashes: "${MISSING_ENV:----:---}"
+DefaultForMissingAndEmpty with missing env and empty default: "${MISSING_ENV:-}"
+DefaultForMissingAndEmpty with empty env: "${EMPTY_ENV:-default}"
+DefaultForMissingAndEmpty with empty env and many dashes: "${EMPTY_ENV:----:---}"
+DefaultForMissingAndEmpty with empty env and empty default: "${EMPTY_ENV:-}"
```

Comment:
Now we support the following options of vars interpolation:
<img width="688" alt="image" src="https://github.com/user-attachments/assets/dbc5246a-3505-4c48-9978-589834908664">

The whole list is here: https://docs.docker.com/compose/how-tos/environment-variables/variable-interpolation/#interpolation-syntax